metals/src/main/scala/scala/meta/internal/metals/Ammonite.scala

```diff
@@ -0,0 +1,390 @@
+      catch {
+        case NonFatal(_) =>
+      }
+      try buildShutdown.asJava.get(100, TimeUnit.MILLISECONDS)
+      catch {
+        case _: TimeoutException =>
+      }
+    }
+  }
+
+  def loaded(path: AbsolutePath): Boolean =
+    path.isAmmoniteScript && {
+      val uri = path.toNIO.toUri.toASCIIString
+      lastImportedBuild0.targetUris.contains(uri)
+    }
+
+  private def connectToNewBuildServer(
+      build: BuildServerConnection
+  ): Future[BuildChange] = {
+    scribe.info(s"Connected to Ammonite Build server v${build.version}")
+    cancelables.add(build)
+    compilers().cancel()
+    buildServer0 = Some(build)
+    val importedBuild0 = // FIXME timed("imported ammonite build") {
+      MetalsLanguageServer.importedBuild(build)
+    for {
+      build0 <- statusBar().trackFuture(
+        "Importing Ammonite scripts",
+        importedBuild0
+      )
+      _ = {
+        lastImportedBuild0 = build0
+      }
+      _ <- indexWorkspace()
+      // _ = checkRunningBloopVersion(build.version)
+      _ <- Future.sequence[Unit, List](
+        compilations()
+          .cascadeCompileFiles(buffers.open.toSeq) ::
+          compilers().load(
+            buffers.open.toSeq.filter(!_.isAmmoniteScript)
+          ) ::
+          Nil
+      )
+    } yield BuildChange.Reconnected
+  }
+
+  private def disconnectOldBuildServer(): Future[Unit] = {
+    if (buildServer0.isDefined)
+      scribe.info("disconnected: ammonite build server")
+    buildServer0 match {
+      case None => Future.successful(())
+      case Some(value) =>
+        // remove from cancellable too?
+        buildServer0 = None
+        lastImportedBuild0 = ImportedBuild.empty
+        cancelables.remove(value)
+        diagnostics().resetAmmoniteScripts()
+        value.shutdown()
+    }
+  }
+
+  def maybeImport(path: AbsolutePath): Unit =
+    if (path.isAmmoniteScript && !loaded(path))
+      languageClient()
+        .showMessageRequest(Messages.ImportAmmoniteScript.params())
+        .asScala
+        .onComplete {
+          case Failure(e) => scribe.warn("Error requesting Ammonite import", e)
+          case Success(resp) =>
+            resp.getTitle match {
+              case Messages.ImportAmmoniteScript.doImport =>
+                start(Some(path)).asScala.onComplete {
+                  case Failure(e) =>
+                    scribe.warn(s"Error importing Ammonite script $path", e)
+                  case Success(_) =>
+                }
+              case _ =>
+            }
+        }
+
+  def start(): CompletableFuture[Object] =
+    start(None)
+
+  def start(doc: Option[AbsolutePath]): CompletableFuture[Object] = {
+
+    disconnectOldBuildServer().onComplete {
+      case Failure(e) =>
+        scribe.warn("Error disconnecting old Ammonite build server", e)
+      case Success(()) =>
+    }
+
+    // TODO Look at buffers.open rather than focusedDocument?
+    val commandScriptOpt = doc.orElse(focusedDocument()) match {
```

Comment:
If someone starts AmmoniteServer, they might want to start them all. Relying on whether it's focused doesn't seem that useful for me.

Reply:
`focusedDocument` is only used to read the scala and Ammonite version (users can put a comment like `// scala 2.13.1, ammonite 2.0.4` at the top of a script), and only so when users use the `ammonite-start` command directly (when they click "Import" in the "Ammonite script detected" dialog, the script that trigerred the dialog is used instead).

Apart from that, it's not used.

When we start Ammonite, we pass it all the scripts from `buffers.open`. Then, Ammonite loads those, along with their script dependencies. So what is loaded at the end is all opened scripts plus their transitive script dependencies.